### DISCUSSION TITLE
Is it possible to backtest dollar cost averging (DCA) and/or portfolio rebalancing strategies (for crypto currencies)?

### DISCUSSION THREAD

**Discussion by d4fuQQ** [kernc/backtesting.py#653] [Q&A]
Hi,
I was wondering if it's possible to run backtesting scenarios (specific examples below) that allow for recurring investments and perhaps even for several crypto currencies at the same time?
In addition, I was wondering if a backtest could be set up in a way that a portfolio is rebalanced daily (or weekly), e.g. for a simple 80/20% split portfolio consisting of ETH/USD.

Below are a few examples for which I would like to try backtesting:

...regarding **DCA only**:

1. what day of the week is best to invest in regularly (*🠔 pick the best day as your benchmark for DCA and compare the results with the additional backtesting scenarios below*)
2. how do certain combinations of coins perform

...  for **rebalancing only**:

1. what's the return when using an 80/20 split portfolio with ETH/USD (daily rebalancing)
2. what's the return when using an 80/20 split portfolio with ETH/USD (weekly rebalancing)
3. what would be the return when rebalancing a 40/40/20 BTC/ETH/USD portfolio (daily)
4. what would be the return when rebalancing a 40/40/20 BTC/ETH/USD portfolio (weekly)

... and for combining **DCA + rebalancing**:

1. what's the return when adding daily rebalance to an 80/20 split ETH/USD portfolio that gets new funds every week through DCA
2. what's the return when adding weekly rebalance to an 80/20 split ETH/USD portfolio that gets new funds every week through DCA
3. what's the return when adding daily rebalance to an 40/40/20 split BTC/ETH/USD portfolio that gets new funds every week through DCA
4. what's the return when adding weekly rebalance to an 40/40/20 split BTC/ETH/USD portfolio that gets new funds every week through DCA

etc.

**Comment by eervin123** [comment]
This doesn't answer all of your questions but may provide some inspiration. Chad Thackray's youtube videos are pretty good. Here is one on Dollar Cost Averaging with backtesting.py https://www.youtube.com/watch?v=Kgr8JjWX0zk&t=36s  Here is the blog. https://greyhoundanalytics.com/blog/backtest-your-dollar-cost-average-strategy-easily-in-python/

Here is his full backtesting.py tutorial. https://greyhoundanalytics.com/blog/backtestingpy-a-complete-quickstart-guide/

Here is the youtube video of that https://www.youtube.com/watch?v=e4ytbIm2Xg0&t=3s
